### DISCUSSION TITLE
Providing Data through middleware to Page. Is that feasible currently?

### DISCUSSION THREAD

**Discussion by vasuneet** [vercel/next.js#37256] [Help]
Hey Team,
Currently I am using the custom server along with NextJS and I am able to pass the data when getting page through custom server like this :

module.exports = (nextapp, nextServerOptions, proxyConfiguration) => {
  (async () => {
    await initialDataPopulation(); // In this I am saving some data in cache object with key name data i.e. into cache["data"]
    myData=cache["data"]; 
    nextapp.prepare().then(() => {
        app.get(`*`, (req, res) => {
              const path = req.path;
              var currentRoute = path.replace(/^\/|\/$/g, '').split("/").pop();
              return nextapp.render(req, res, `/` + currentRoute, {
                                    amp: { myData }, //here I am appending data in pagecontext
            });
        }

     }
}}

But If I am avoiding custom server and want to go with middleware. I am not seeing any way to do the same. I can pass data through cookie however which I don't want.

**Comment by Korayem** [comment]
You can't
See similar question: https://github.com/vercel/next.js/discussions/36904
